src/Controller/LoginController.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace App\Controller;
+
+use ItkDev\OpenIdConnectBundle\Exception\InvalidProviderException;
+use ItkDev\OpenIdConnectBundle\Security\OpenIdConfigurationProviderManager;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\Form\Extension\Core\Type\CheckboxType;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Http\Util\TargetPathTrait;
+use Symfony\Component\Translation\TranslatableMessage;
+
+class LoginController extends AbstractController
+{
+    use TargetPathTrait;
+
+    private OpenIdConfigurationProviderManager $providerManager;
+
+    public function __construct(OpenIdConfigurationProviderManager $providerManager)
+    {
+        $this->providerManager = $providerManager;
+    }
+
+    /**
+     * @Route("/login", name="login")
+     */
+    public function index(Request $request, SessionInterface $session): Response
+    {
+        $authenticationProvider = null;
+
+        // Look for authentication provider (aliased to "role") in query string.
+        if (null !== $request->query->has('role')) {
+            $authenticationProvider = $request->query->get('role');
+        }
+
+        // Look for authentication provider (aliased to "role") in target path query string.
+        if (null === $authenticationProvider) {
+            $targetPath = $this->getTargetPath($request->getSession(), 'main');
```

Comment:
The firewall name could be configurable instead of relying on a constant.

Reply:
https://github.com/itk-dev/naevnssekretariatet/pull/75/commits/5db766241a47a02c9c084f29967348c60539e756